packages/smart-contracts/hardhat.config.ts

```diff
@@ -66,52 +67,52 @@ export default {
       accounts,
     },
     matic: {
-      url: process.env.WEB3_PROVIDER_URL || 'https://polygon-rpc.com',
+      url: process.env.WEB3_PROVIDER_URL || utils.networkRpcs.matic,
```

Comment:
Would be nice to factorize a bit:
```suggestion
// Once and for all
      const url = (network) => process.env.WEB3_PROVIDER_URL || utils.networkRpcs[network]
// For every network
      url: url(matic)
```